.github/workflows/makefile-lint.yml

```diff
@@ -0,0 +1,96 @@
+# .github/workflows/makefile-lint.yml
+---
+name: Makefile Lint
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches: ["main", "master", "stable"]
+  pull_request:
+    branches: ["main", "master", "stable", "feature-*", "patch-*", "HOTFIX-*"]
+
+permissions: {}  # Setting default permissions to none for enhanced security
+
+jobs:
+  makefile-lint:
+    permissions:
+      contents: read
+      statuses: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Install Apt-Get Dependencies
+        run: |
+          sudo apt-get update || exit 1
+          sudo apt-get install -y yamllint golang-go pandoc || exit 1
+          ERR_MSG_STUB="Go installation failed."
+          ERR_DETAILS="file=.github/workflows/makefile-lint.yml,line=25,endLine=30,title=FAILURE"
+          ERR_MSG_LINE="::error ${ERR_DETAILS}::ERROR ${ERR_MSG_STUB}"
+          go version || { printf "%s\n" "${ERR_MSG_LINE}"; exit 1; }
+      - name: Lint Workflow YAML
+        if: ${{ success() }}
+        run: |
+          yamllint ${{ vars.YAML_ARGS }} .github/workflows/makefile-lint.yml
+      - name: Install checkmake
+        if: ${{ success() }}
+        env:
+          BUILDER_NAME: "makefile-lint"
+          BUILDER_EMAIL: "no-reply@localhost"
+        run: |
+          git clone --depth 1 --branch main https://github.com/mrtazz/checkmake.git checkmake
+          cd checkmake
+          EXPECTED_SHA="bd26d7905e47713ff0bf3b0e5e7b9c55f0d24e53" # v0.2.2+8
+          ACTUAL_SHA=$(git rev-parse HEAD)
+          if [ "$EXPECTED_SHA" != "$ACTUAL_SHA" ]; then
+            ERR_MSG_STUB="Checkmake repository hash verification failed."
+            ERR_DETAILS="file=.github/workflows/makefile-lint.yml,line=42,endLine=51,title=FAILURE"
+            ERR_MSG_LINE="::error ${ERR_DETAILS}::ERROR ${ERR_MSG_STUB}"
+            printf "%s\n" "${ERR_MSG_LINE}"
+            exit 1;
+          fi
+          make || { \
+            ERR_MSG_STUB="Checkmake build failed." \
+            ERR_DETS="file=.github/workflows/makefile-lint.yml,line=52,endLine=59,title=FAILURE" \
+            ERR_MSG_LINE="::error ${ERR_DETS}::ERROR ${ERR_MSG_STUB}" \
+            printf "%s\n" "${ERR_MSG_LINE}" \
+            exit 1 \
+             ;}
```

Comment:
open a followup GH issue for this work. it will be performed on a development branch first, and released with another future PR/MR. be sure to include the implementation code in the new issue's description.